app/components/Navbar.tsx

```diff
@@ -35,14 +40,23 @@ const Navbar = () => {
     };
   }, []);
 
+  useEffect(() => {
+    const newIcons = {
+      menuIcon: theme === "light" ? menuButton : menuButtonDarkMode,
+      downloadIcon: theme === "light" ? download : downloadDarkMode,
+      lightOrDarkModeIcon: theme === "light" ? lightModeIcon : darkModeIcon,
+    };
+    setIcons(newIcons);
+  }, [theme]);
+
```

Comment:
Not necessarily required, but try to define the icon mappings outside the useEffect to make the function easier to read and understand.

```
const iconMappings = {
  menuIcon: { light: menuButton, dark: menuButtonDarkMode },
  downloadIcon: { light: download, dark: downloadDarkMode },
  lightOrDarkModeIcon: { light: lightModeIcon, dark: darkModeIcon },
};

useEffect(() => {
  const newIcons = Object.fromEntries(
    Object.entries(iconMappings).map(([key, value]) => [key, value[theme]])
  );
  setIcons(newIcons);
}, [theme]);
```